modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryDataRegion.java

```diff
@@ -17,24 +17,24 @@
 
 package org.apache.ignite.internal.storage.pagememory;
 
-import java.util.Objects;
 import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.pagememory.DataRegion;
 import org.apache.ignite.internal.pagememory.PageMemory;
-import org.apache.ignite.internal.pagememory.PageMemoryDataRegion;
-import org.apache.ignite.internal.pagememory.configuration.schema.PageMemoryDataRegionConfiguration;
+import org.apache.ignite.internal.pagememory.configuration.schema.BasePageMemoryDataRegionConfiguration;
 import org.apache.ignite.internal.pagememory.io.PageIoRegistry;
+import org.apache.ignite.internal.storage.StorageException;
 
 /**
  * Abstract data region for {@link PageMemoryStorageEngine}. Based on a {@link PageMemory}.
  */
-abstract class AbstractPageMemoryDataRegion implements PageMemoryDataRegion, IgniteComponent {
-    protected final PageMemoryDataRegionConfiguration cfg;
+abstract class AbstractPageMemoryDataRegion<T extends PageMemory> implements DataRegion<T>, IgniteComponent {
```

Comment:
I don't understand why it's bad?

Reply:
Because it breaks the intention of this interface. IgniteComponents have their own lifecycle and are managed by the `LifecycleManager`. Anyway, why do you need an interface to have `start` and `stop` methods? What's wrong with declaring them in this class directly?

Reply:
You're right, I'll fix it.